lib/libzfs/os/linux/libzfs_pool_os.c

```diff
@@ -284,12 +286,108 @@ zpool_label_disk(libzfs_handle_t *hdl, zpool_handle_t *zhp, const char *name)
 	 * etc. were all pretty specific.  V_USR is as close to reality as we
 	 * can get, in the absence of V_OTHER.
 	 */
-	vtoc->efi_parts[0].p_tag = V_USR;
-	zpool_label_name(vtoc->efi_parts[0].p_name, EFI_PART_NAME_LEN);
+	/* first fix the partition start block */
+	if (start_block == MAXOFFSET_T)
+		start_block = NEW_START_BLOCK;
 
-	vtoc->efi_parts[8].p_start = slice_size + start_block;
-	vtoc->efi_parts[8].p_size = resv;
-	vtoc->efi_parts[8].p_tag = V_RESERVED;
+	/*
+	 * EFI System partition is using slice 0.
+	 * ZFS is on slice 1 and slice 8 is reserved.
```

Comment:
can we finally get rid of the old Sun manufacturing slice 8? it has nothing to do with ZFS

Reply:
The disk expansion code in `lib/libefi/rdwr_efi.c` which is needed when doing an expansion with `zpool online -e <pool> <disk>` (which is a feature I care about) seems to currently also depend on the existence of that reserved partition.
Also, unfortunately in direct tension with this PR as written is the fact that the linux specific `zfs_append_partition` in `lib/libzutil/os/linux/zutil_device_path_os.c` currently seems to assume that the pool is on always partition 1.

So in principle, "sure." In practice, maybe that should be a separate PR? Also, I'm not sure how much backward compatibility we want to retain for expanding pools both with and without that extra partition. Maintaining the backward compatibility might be harder than keeping the partition around on new pools.